pkgs/servers/sql/postgresql/patches/empty-pg-config-view-15+.patch

```diff
@@ -0,0 +1,54 @@
+Empty the pg_config system information view. This view keeps references to 
+several -dev outputs, which we want to avoid to keep closure size down.
+
+The alternative to this patch would be to nuke references across the board,
+but this will also affect the output of the pg_config utility. By emptying
+the view only, we keep the pg_config binary intact. It resides in the -dev
+output, so it's fine to have all those references there.
+
+---
+--- a/src/backend/utils/misc/pg_config.c
++++ b/src/backend/utils/misc/pg_config.c
+@@ -32,20 +32,5 @@ pg_config(PG_FUNCTION_ARGS)
+ 	/* initialize our tuplestore */
+ 	InitMaterializedSRF(fcinfo, 0);
+ 
+-	configdata = get_configdata(my_exec_path, &configdata_len);
```

Comment:
I wonder if it would be feasible to compile nuked references in here only, i.e. pg_config returns the correct stuff, but this view returns nuked references only.

But I'm not aware of any uses of this, so I don't know if this even makes sense.
I think with good documentation I'm fine and a more sophisticated solution can be added in further iterations if this is actually requested by people.